terraform/aptos-node/aws/README.md

```diff
@@ -0,0 +1,181 @@
+  ```
+
+5. Initialise Terraform in the same directory of your `main.tf` file
+  ```
+  $ terraform init
+  ```
+This should download all the terraform dependencies for you, in the `.terraform` folder.
+
+6. Create a new Terraform workspace to isolate your environments:
+  ```
+  $ terraform workspace new $WORKSPACE
+  # This command will list all workspaces
+  $ terraform workspace list
+  ```
+
+7. Apply the configuration.
+  ```
+  $ terraform apply
+  ```
+  This might take a while to finish (~20 minutes), Terraform will create all the resources on your cloud account.
+
+8. Once terraform apply finishes, you can check if those resources are created:
+
+    - `aws eks update-kubeconfig --name aptos-$WORKSPACE` to configure access for your k8s cluster.
+    - `kubectl get pods` this should have haproxy, validator and fullnode. with validator and fullnode pod `pending` (require further action in later steps)
+    - `kubectl get svc` this should have `validator-lb` and `fullnode-lb`, with an external-IP you can share later for connectivity.
+
+9. Get your node IP info:
+
+    ```
+    $ export VALIDATOR_ADDRESS="$(kubectl get svc ${WORKSPACE}-aptos-node-validator-lb --output jsonpath='{.status.loadBalancer.ingress[0].hostname}')"
+
+    $ export FULLNODE_ADDRESS="$(kubectl get svc ${WORKSPACE}-aptos-node-fullnode-lb --output jsonpath='{.status.loadBalancer.ingress[0].hostname}')"
+    ```
+
+10. Generate key pairs (node owner key, consensus key and networking key) in your working directory.
+
+    ```
+    $ aptos genesis generate-keys --output-dir ~/$WORKSPACE
+    ```
+
+    This will create a `private-keys.yaml` file for you, Backup your key files somewhere safe, this is important for you to establish ownership of your node, and it will be used to claim your rewards later if eligible. Very important!!
+
+11. Configure validator information.
+
+    ```
+    $ aptos genesis set-validator-configuration --keys-dir ~/$WORKSPACE --local-repository-dir ~/$WORKSPACE --username <pick a username for your node> --validator-host $VALIDATOR_ADDRESS:6180 --full-node-host $FULLNODE_ADDRESS:6182
+
+    ```
+
+    This will create a YAML file in your working directory with your username, e.g. `aptosbot.yml`, it should looks like:
+
+    ```
+    ---
+    account_address: 7410973313fd0b5c69560fd8cd9c4aaeef873f869d292d1bb94b1872e737d64f
+    consensus_key: "0x4e6323a4692866d54316f3b08493f161746fda4daaacb6f0a04ec36b6160fdce"
+    account_key: "0x83f090aee4525052f3b504805c2a0b1d37553d611129289ede2fc9ca5f6aed3c"
+    network_key: "0xa06381a17b090b8db5ffef97c6e861baad94a1b0e3210e6309de84c15337811d"
+    validator_host:
+      host: 30247cc34f270cb8.elb.us-west-2.amazonaws.com
+      port: 6180
+    full_node_host:
+      host: abc5b9734d4cc418.elb.us-west-2.amazonaws.com
+      port: 6182
+    stake_amount: 1
+    ```
+
+12. Create layout YAML file, which defines the node in the validatorSet, for test mode, we can create a genesis blob containing only one node.
```

Comment:
Create layout YAML file, which defines the node in the validatorSet. For test mode, we can create a genesis blob containing only one node.